x-pack/plugins/upgrade_assistant/server/lib/telemetry/es_ui_open_apis.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+import {
+  UIOpen,
+  UIOpenOption,
+  UPGRADE_ASSISTANT_DOC_ID,
+  UPGRADE_ASSISTANT_TYPE,
+  UpgradeAssistantTelemetryServer,
+} from '../../../common/types';
+
+async function incrementUIOpenOptionCounter(
+  server: UpgradeAssistantTelemetryServer,
+  uiOpenOptionCounter: UIOpenOption
+) {
+  const { getSavedObjectsRepository } = server.savedObjects;
+  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
```

Comment:
I think we want callWithRequest here?  this is triggered from an api endpoint right?  //cc @elastic/kibana-security

Reply:
Is there a reason we're using the SavedObjectsRepository here, instead of the SavedObjectsClient? Using the client, you'll get auth checks for free.

We don't want to use a raw `callWithRequest` here, because end users may not have direct access to the .kibana index anymore.

Do we want these metrics recorded for read-only kibana users as well as full kibana users?

Reply:
if you get a chance can you explain the difference between repository and client?

Reply:
The repository is a lower level utility than the client. The repository is really only concerned with reading/writing documents to/from Elasticsearch. It's responsible for transforming the raw documents into what we think of as Saved Objects, and it's also responsible for going the other direction: transforming our Saved Objects into raw documents.

The Saved Objects Client uses the repository to do that low-level work, but the client itself provides additional pluggable functionality via middleware/wrappers. Using the client will automatically enforce authorization rules for you, and it will automatically handle space-aware and space-agnostic types without the consumer having to give that any thought.

Using the repository, you are responsible for doing _all_ of that extra yourself, which isn't advised.

Using the repository _might_ be fine here, it just depends on our needs. Do we want these metrics recorded for read-only users of Kibana? If not, then we can't use the repository. I don't know enough about the assistant to know if it's beneficial for read-only users, or if we want those hits recorded.

Using the repository will record these metrics for any authenticated Kibana user who hits the endpoint, regardless of their access rights.

Reply:
@jbudz after chat with @legrego we end in an agreement that using the the lower level repository with `callWithInternalUser` it's fine and  it isn’t really posing a security threat. If in the future we implement the `incrementCounter` function into the `SavedObjectsClient` we can migrate this could to just `request.getSavedObjectsClient().incrementCounter`.

Also mention to note that we have also chat with @joshdover to understand how the upgrade assistant itself is handling permissions.